tests/sample/map.c

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// clang -O2 -Werror -c map_reuse.c -o map_reuse.o
+//
+// For bpf code: clang -target bpf -O2 -Werror -c map_reuse.c -o map_reuse.o
+// this passes the checker
+
+#include "bpf_helpers.h"
+#include "ebpf.h"
+
+#define DECLARE_MAP(TYPE)                              \
+    SEC("maps")                                        \
+    struct _ebpf_map_definition_in_file TYPE##_map = { \
+        .type = BPF_MAP_TYPE_##TYPE,                   \
+        .key_size = sizeof(uint32_t),                  \
+        .value_size = sizeof(uint32_t),                \
+        .max_entries = 10,                             \
+    };
+
+#define TEST_MAP(TYPE, TEST)                                                \
+    result = test_##TEST##_map(&TYPE##_map);                                \
+    if (result < 0) {                                                       \
+        bpf_printk("Test " #TEST " for map " #TYPE " returned %d", result); \
+        return result;                                                      \
+    }
+
+inline int
+test_GENERAL_map(struct _ebpf_map_definition_in_file* map)
+{
+    uint32_t key = 0;
+    uint32_t value = 1;
+    uint32_t* return_value = NULL;
+    int result = 0;
+    result = bpf_map_update_elem(map, &key, &value, BPF_ANY);
+    if (result < 0) {
+        bpf_printk("bpf_map_update_elem returned %d", result);
+        return result;
+    }
+
+    return_value = bpf_map_lookup_elem(map, &key);
+    if (return_value == NULL) {
+        bpf_printk("bpf_map_lookup_elem returned NULL");
+        return -1;
+    }
+
+    result = bpf_map_delete_elem(map, &key);
+    if (result < 0) {
+        bpf_printk("bpf_map_delete_elem returned %d", result);
+        return result;
+    }
+
+    return 0;
+}
+
+inline int
+test_LRU_map(struct _ebpf_map_definition_in_file* map)
+{
+    uint32_t key = 0;
+    uint32_t value = 1;
+    int result;
+
+    // Insert capacity + 1 entries
+    for (key = 0; key < 11; key++) {
+        result = bpf_map_update_elem(map, &key, &value, BPF_ANY);
+        if (result < 0) {
+            bpf_printk("bpf_map_update_elem returned %d", result);
+            return result;
+        }
+    }
+    return 0;
+}
+
+inline int
+test_PUSH_POP_map(struct _ebpf_map_definition_in_file* map)
+{
+    uint32_t value = 1;
+    int result;
+
+    result = bpf_map_push_elem(map, &value, 0);
+    if (result < 0) {
+        bpf_printk("bpf_map_push_elem returned %d", result);
+        return result;
+    }
+
+    result = bpf_map_peek_elem(map, &value);
+    if (result < 0) {
+        bpf_printk("bpf_map_peek_elem returned %d", result);
+        return result;
+    }
+
```

Comment:
This doesn't actually test that the value returned is the value pushed.
Also it doesn't test order.  Please add tests.
E.g., for stack:
push 1
push 2
pop (should be 2)
pop (should be 1)